FormalConjectures/ErdosProblems/274.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 274
+
+*References:*
+[erdosproblems.com/274](https://www.erdosproblems.com/274)
+[Wikipedia](https://en.wikipedia.org/wiki/Herzog%E2%80%93Sch%C3%B6nheim_conjecture)
+-/
+
+open scoped Pointwise
+
+/--
+If `G` is an abelian group then can there exist an exact covering of `G` by more than one cosets of
+different sizes? (i.e. each element is contained in exactly one of the cosets.)
+-/
+@[category research open, AMS 20]
+theorem erdos_274 :
+    (∀ (G : Type*), [CommGroup G] →
+    ∃ (P : Partition (⊤ : Subgroup G)), 1 < P.parts.ncard ∧
+      (∀ A ∈ P.parts, ∃ (s : G) (H : Subgroup G), s • (H : Set G) = A) ∧
+      P.parts.Pairwise fun A B ↦ Nat.card A ≠ Nat.card B) ↔ answer(sorry) := by
```

Comment:
I think you're in trouble here since you're using `Nat.card` without any finitness assumptions on the subgroups.

Reply:
I think not because groups always have size \ge 1. So size is infinite iff Nat.card = 0. But maybe I should still change it because this is a "trick".

Reply:
I think here the issue is that the condition prevents you from having e.g two infinite subgroups of different cardinalities (since they both have Nat.card 0)

Reply:
Ahh for some reason I thought this problem was not really concerned about larger cardinalities but maybe it is actually.

Reply:
Okay I just came back to this now. Do you think it's in the spirit of the problem to care about different sized infinite cardinalities? I'm mentioning this because if I choose to formulate this using cardinalities then we lose the `Subgroup.index` API, as I can't find a corresponding function valued in Cardinals.